I forgot to pull the changes on master before doing these changes, sorry. I wanted to fix this, but then I realized that the running example fails to build on master. There seems to be an incomplete change putting things in `.build`?

```
$ make .build/io_program_exe                                                                        (git)-[master] 
fstar.exe --unsafe_tactic_exec --z3version 4.15.3 --cache_checked_modules --cache_dir .build/cache --record_hints --use_hints --hint_dir .build/hints --odir .build/odir --include lambdabox --warn_error -241-247-274-321-333 --silent  lambdabox/LambdaBoxExamples.fst
opam exec --switch=only-peregrine -- peregrine ocaml .build/io_program.ast -o .build/io_program_raw.mlf
peregrine: FILE argument: no '.build/io_program.ast' file or directory
Usage: peregrine ocaml [--opt] [--typed=VAL] [OPTION]… FILE
Try 'peregrine ocaml --help' or 'peregrine --help' for more information.
make: *** [Makefile:116: .build/io_program_raw.mlf] Error 124
```

--- a/seiostar/lambdabox/axioms.ml
+++ b/seiostar/lambdabox/axioms.ml
@@ -59,10 +59,11 @@ let def_Runtime_io_read fd =
       let rec read_line () =
         let got = Unix.read ufd b 0 1 in
         if got = 0 then ()
-        else
+        else begin
           let c = Bytes.get b 0 in
-          if c = '\n' then ()
-          else begin Buffer.add_char buf c; read_line () end
+          Buffer.add_char buf c;
+          read_line ()
+        end
       in
       read_line ();
       make_inl (Obj.repr (Buffer.contents buf))

--- a/seiostar/lambdabox/axioms.ml
+++ b/seiostar/lambdabox/axioms.ml
@@ -79,6 +79,10 @@ let def_Runtime_io_write (fd : Obj.t) (msg : Obj.t) : Obj.t =
       let b = Bytes.of_string s in
       let len = Bytes.length b in
       let _ = Unix.write ufd b 0 len in
+      (* truncate up to current pos *)
+      let pos = Unix.lseek ufd 0 Unix.SEEK_CUR in
+      Unix.ftruncate ufd pos;
+
       make_inl unit_val
     with _ -> make_inr unit_val)
 

